packages/gatsby/src/query/graphql-runner.js

```diff
@@ -18,23 +20,96 @@ class GraphQLRunner {
     })
   }
 
-  query(query, context) {
+  async query(query, context, { parentSpan } = {}) {
     const { schema, schemaCustomization } = this.store.getState()
 
-    return graphql(
+    const activity = new SpanTracker(`GraphQL Query`, {
```

Comment:
I would rename to spanTracker as it collides with createActivity.

```suggestion
    const spanTracker = new SpanTracker(`GraphQL Query`, {
```